dbcon/joblist/tuple-bps.cpp

```diff
@@ -2318,6 +2325,17 @@ void TupleBPS::receiveMultiPrimitiveMessages(uint32_t threadID)
                                 generateJoinResultSet(joinerOutput, joinedBaseRow, smallMappings,
                                                       0, local_outputRG, joinedData, &rgDatav, smallSideRows, postJoinRow);
 
+                                // MCOL-3879.  JoinedBaseRow was originally an in-line row because of the
+                                // way its used here, as scratch space.  Values are constantly replaced.  
+                                // That was fine when columns were limited to 8KB.  Now they can be 2GB.  :D  
+                                // Unfortunately values don't get replaced in StringTables, they accumulate.  
+                                // So here, we need to check whether we've accumulated 'too much' memory and 
+                                // if so, reinit the string storage.
+                                if (joinedBaseRow.usesStringTable()&& joinedBaseRowData.getStringTableMemUsage() > 50 * (1 << 20)) 
+                                {
+                                    joinedBaseRowData.clearStringStore();
+                                }
```

Comment:
We should move this threshold value (50 * (1 << 20)) into a common location (maybe in rowgroup.h) as a const int or a macro, so it can be easily updated later.

Use of UNLIKELY around the if predicate might be a suitable optimization?